server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -5856,4 +5888,37 @@ public void testPruneAwayDeletedButRetainedIds() throws Exception {
             }
         }
     }
+
+    public void testRecoverFromLocalTranslog() throws Exception {
```

Comment:
This test fails on master and 7.x (1 of 5000).

Reply:
cool can we try to make a test that fails more often?

Reply:
The test fails when we have out of order in the safe commit and a merge prunes _id. This test index/delete/flush/merge randomly then restarting an engine should yield the same documents as before. I can make it fail more happen but I prefer to leave as it is (hope we can catch other situations). Are you okay?

Reply:
are there other tests that are failing more often than 1 in 5000, to make sure that this change has enough coverage?

Reply:
Yes, I can add that test in a follow-up.